src/logic/map_objects/immovable.cc

```diff
@@ -115,6 +120,11 @@ void BaseImmovable::unset_position(EditorGameBase& egbase, const Coords& c) {
 
 	assert(f.field->immovable == this);
 
+	// Needed so players can update partially visible buildings' vision.
+	if (get_size() >= BIG) {
+		Notifications::publish(NoteFieldTerrainChanged{f, map->get_index(f)});
+	}
+
 	f.field->immovable = nullptr;
```

Comment:
Shouldn't the field rather be rediscovered *after* unlinking the immovable from the field?

Reply:
Right, good point. LGTM :)